pkg/asset/agent/mirror/registriesconf.go

```diff
@@ -139,14 +139,16 @@ func (i *RegistriesConf) Generate(dependencies asset.Parents) error {
 	releaseImage := &releaseimage.Image{}
 	dependencies.Get(installConfig, releaseImage)
 
-	if !installConfig.Supplied || len(installConfig.Config.ImageContentSources) == 0 {
+	if !installConfig.Supplied || len(installConfig.Config.ImageContentSources) == 0 && len(installConfig.Config.ImageDigestSources) == 0 {
 		return i.generateDefaultRegistriesConf()
 	}
 
 	registries := &sysregistriesv2.V2RegistriesConf{
 		Registries: []sysregistriesv2.Registry{},
 	}
-	for _, group := range bootstrap.MergedMirrorSets(installConfig.Config.ImageContentSources) {
+	digestMirrorSources := bootstrap.ContentSourceToDigestMirror(installConfig.Config.ImageContentSources)
+	digestMirrorSources = append(digestMirrorSources, installConfig.Config.ImageDigestSources...)
```

Comment:
below you have validation logic that rejects installconfigs that set both fields, so why here do you have logic that merges the two fields?

is it to be treated as invalid, or not?

Reply:
@bparees  It is invalid. In practice, it should fail before this line or one of the fields should be empty.

Reply:
if it is invalid, then why does the code here treat it as valid?  i'd expect this code to throw an error if we manage to get here with both fields populated (which, as you say, we should not).

Reply:
I thought the `ValidateInstallConfig` is called before this. I updated the validation code here anyway.